src/api/app/jobs/project_status_calculator.rb

```diff
@@ -131,26 +129,29 @@ def parse_jobhistory(dname, repo, arch)
 
     uri = '/build/%s/%s/%s/_jobhistory?code=lastfailures' % [CGI.escape(dname), CGI.escape(repo), arch]
 
-    ret = []
     d = Suse::Backend.get(uri).body
-    unless d.blank?
-      data = Xmlhash.parse(d)
 
-      data.elements('jobhist') do |p|
+    return [] if d.blank?
+
+    data = Xmlhash.parse(d)
+
+    data.elements('jobhist').map do |p|
         line = {'name' => p['package'],
```

Comment:
very careful here - there is a difference if you can elements with a block or as such - as you talk about memory usage